codegen/src/types.rs

```diff
@@ -749,24 +751,24 @@ impl Message {
         Ok(())
     }
 
-    fn set_package(&mut self, package: &str) {
+    fn set_package(&mut self, package: &str, module: &str) {
```

Comment:
It's a hack, and so needs review. Since `mod_` is not prefixed to module names, I needed a way of ensuring that _nested_ messages did actually sit in a `mod_MESSAGE` module. Hence the separate package (`a.b.c.C.D`) vs module (`a.b.c.mod_C.D`).